### PR TITLE
fix #20883 Unspecified generic on default value segfaults the compiler

### LINKED ISSUE #20883: Unspecified generic on default value segfaults the compiler
### What happened?

```nim
proc foo*[U](x: U = U(1e-6)) =
  echo x
  
foo()
```
Not providing the type `U` causes a segmentation fault in the compiler instead of giving an error. Removing the `U(...)` around `1e-6` works, though. 

### Nim Version

Nim Compiler Version 1.7.3 [Linux: amd64]
Compiled at 2022-11-19
Copyright (c) 2006-2022 by Andreas Rumpf

git hash: fe43f751eb9a83f84cc93aa0d752c3658232002d
active boot switches: -d:release

### Current Standard Output Logs

```shell
Segmentation fault (core dumped)
```

### Expected Standard Output Logs

```shell
That the compiler gives an error about the unspecified generic.
```

### Possible Solution

_No response_

### Additional Information

This segfaults on all versions of Nim. (tested back to v1.0)

### PR DESCRIPTION
**Pull request by bung87**
fix #20883

<s>seem can't give more clear message here.</s>

### CODE REVIEW

**Comment by Araq** [comment]
The segfault is bad of course but the example program should simply work and not be rejected.

**Comment by bung87** [comment]
CI fails unrelated.

File: tests/misc/t20883.nim [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  action: reject
+  errormsg: "type mismatch: got <float64> but expected 'typeof(U(0.000001))'"
+  line: 8
+  column: 22
+"""
+
+proc foo*[U](x: U = U(1e-6)) =
+  echo x
+
+foo[float]()
```
Comment (Araq) [inline tests/misc/t20883.nim:11]: This one is valid.

File: tests/misc/t20883.nim [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+foo()
```
Comment (Araq) [inline tests/misc/t20883.nim:12]: This one isn't.

File: tests/misc/t20883.nim (hunk 1 above)
Comment (bung87) [reply tests/misc/t20883.nim:11]: yeah, it's here just ensure am not breaking valid code.

File: tests/misc/t20883.nim (hunk 2 above)
Comment (bung87) [reply tests/misc/t20883.nim:12]: yeah, the full stacktrace show the error starting from line 12, as generic proc should Initialization, so the error line is 8

**Comment by Araq** [comment]
Too risky to merge for 2.0 but will be merged directly afterwards.

**Comment by bung87** [comment]
@Araq since v2 released, it's safe to merge .?

**Pull request merged**